### PR TITLE
partners[openai]: add tests for secret_str for keys

### LINKED ISSUE #12165: For New Contributors: Use SecretStr for api_keys
Hello LangChain community,

We're always happy to see more folks getting involved in contributing to the LangChain codebase.

This is a good first issue if you want to learn more about how to set up
for development in the LangChain codebase.

## Goal

Your contribution will make it safer to print out a LangChain object
without having any secrets included in raw format in the string representation.

## Set up for development

Prior to making any changes in the code:

https://github.com/langchain-ai/langchain/blob/master/.github/CONTRIBUTING.md

Make sure you're able to test, format, lint from `langchain/libs/langchain`

```sh
make test
```

```sh
make format
```

```sh
make lint
```

## Shall you accept

Shall you accept this challenge, please claim one (and only one) of the modules from the list
below as one that you will be working on, and respond to this issue.

Once you've made the required code changes, open a PR and link to this issue.

## Acceptance Criteria

- invoking str or repr on the the object does not show the secret key

Integration test for the code updated to include tests that:

- confirms the object can be initialized with an API key provided via the initializer
- confirms the object can be initialized with an API key provided via an env variable

Confirm that it works:

- either re-run notebook for the given object or else add an appropriate test
that confirms that the actual secret is used appropriately (i.e.,`.get_secret_value()`)

**If your code does not use `get_secret_value()` somewhere, then it probably contains a bug!**

## Modules

- [ ] langchain/chat_models/anyscale.py @aidoskanapyanov 
- [ ] langchain/chat_models/azure_openai.py @onesolpark 
- [x] langchain/chat_models/azureml_endpoint.py @fyasla 
- [ ] langchain/chat_models/baichuan.py
- [ ] langchain/chat_models/everlyai.py @sfreisthler 
- [x] langchain/chat_models/fireworks.py @nepalprabin 
- [ ] langchain/chat_models/google_palm.py @faisalt14 
- [ ] langchain/chat_models/javelin_ai_gateway.py
- [ ] langchain/chat_models/jinachat.py
- [ ] langchain/chat_models/konko.py
- [ ] langchain/chat_models/litellm.py
- [ ] langchain/chat_models/openai.py @AnasKhan0607 
- [ ] langchain/chat_models/tongyi.py
- [ ] langchain/llms/ai21.py
- [x] langchain/llms/aleph_alpha.py @slangenbach 
- [ ] langchain/llms/anthropic.py
- [ ] langchain/llms/anyscale.py @aidoskanapyanov 
- [ ] langchain/llms/arcee.py
- [ ] langchain/llms/azureml_endpoint.py
- [ ] langchain/llms/bananadev.py
- [ ] langchain/llms/cerebriumai.py
- [ ] langchain/llms/cohere.py @arunsathiya 
- [ ] langchain/llms/edenai.py @kristinspenc 
- [ ] langchain/llms/fireworks.py
- [ ] langchain/llms/forefrontai.py
- [ ] langchain/llms/google_palm.py @Harshil-Patel28 
- [ ] langchain/llms/gooseai.py
- [ ] langchain/llms/javelin_ai_gateway.py
- [ ] langchain/llms/minimax.py
- [ ] langchain/llms/nlpcloud.py
- [ ] langchain/llms/openai.py @HassanA01 
- [ ] langchain/llms/petals.py @akshatvishu 
- [ ] langchain/llms/pipelineai.py
- [ ] langchain/llms/predibase.py
- [ ] langchain/llms/stochasticai.py
- [x] langchain/llms/symblai_nebula.py @praveenv 
- [ ] langchain/llms/together.py
- [ ] langchain/llms/tongyi.py
- [ ] langchain/llms/writer.py @ommirzaei 
- [ ] langchain/llms/yandex.py

### Motivation

Prevent secrets from being printed out when printing the given langchain object.

### Your contribution

Please sign up by responding to this issue and including the name of the module.

### PR DESCRIPTION
**Pull request by sepiatone**
**Description:** Add tests to check API keys and Active Directory tokens are masked
**Issue:** Resolves #12165 for OpenAI and Azure OpenAI models
**Dependencies:** None

Also resolves #12473 which may be closed.

Additional contributors @alex4321 (#12473) and @onesolpark (#12542)

### CODE REVIEW

File: libs/partners/openai/tests/unit_tests/test_secrets.py [hunk 1]
```diff
@@ -60,3 +65,336 @@ def test_azure_openai_embeddings_secrets() -> None:
...
+
+
+def test_azure_chat_openai_api_key_masked_when_passed_from_env(
+    monkeypatch: MonkeyPatch, capsys: CaptureFixture
+) -> None:
+    """Test that the API key is masked when passed from an environment variable."""
+    monkeypatch.setenv("AZURE_OPENAI_API_KEY", "secret-api-key")
+    monkeypatch.setenv("AZURE_OPENAI_AD_TOKEN", "secret-ad-token")
+    chat_model = AzureChatOpenAI(
+        openai_api_key="secret-api-key",
```
Comment (ccurme) [inline libs/partners/openai/tests/unit_tests/test_secrets.py:90]: Should we remove the secrets from the constructor in this case, to enforce that they're read from the env vars?

File: libs/partners/openai/tests/unit_tests/test_secrets.py [hunk 2]
```diff
@@ -60,3 +65,336 @@ def test_azure_openai_embeddings_secrets() -> None:
...
+    assert (
+        cast(SecretStr, chat_model.azure_ad_token).get_secret_value()
+        == "secret-ad-token"
+    )
+
+
+# azure openai
+def test_azure_openai_api_key_is_secret_string() -> None:
+    """Test that the API key is stored as a SecretStr."""
+    llm = AzureOpenAI(
```
Comment (ccurme) [inline libs/partners/openai/tests/unit_tests/test_secrets.py:148]: (nit) Are we able to parametrize these at all? e.g.,
```python
from typing import Type

import pytest


@pytest.mark.parametrize(
    "model_class", [AzureChatOpenAI, AzureOpenAI, AzureOpenAIEmbeddings]
)
def test_api_key_is_secret_string(model_class: Type) -> None:
    model_instance = model_class(
        openai_api_key="secret-api-key",
        azure_endpoint="endpoint",
        azure_ad_token="secret-ad-token",
        api_version="version",
    )
    assert isinstance(model_instance.openai_api_key, SecretStr)
    ...
```

**Review by ccurme — COMMENTED** [review]
Thank you!

File: libs/partners/openai/tests/unit_tests/test_secrets.py (hunk 1 above)
Comment (sepiatone) [reply libs/partners/openai/tests/unit_tests/test_secrets.py:90]: Yes. I was a tad too aggressive with the copy-paste.

File: libs/partners/openai/tests/unit_tests/test_secrets.py (hunk 2 above)
Comment (sepiatone) [reply libs/partners/openai/tests/unit_tests/test_secrets.py:148]: Yes. Thank you - I'd overlooked this simplification.

**Review by baskaryan — APPROVED** [review]

**Pull request merged**